agent/nmap_wrapper.py

```diff
@@ -24,8 +25,16 @@ def _parse_output(xml_output: str) -> Any:
     Returns:
         dict of the scan's result.
     """
-    parsed_xml = xmltodict.parse(xml_output)
-    return parsed_xml
+    try:
+        parsed_xml: dict[str, Any] = xmltodict.parse(xml_output)
+        return parsed_xml
+    except xml.parsers.expat.ExpatError as parsing_error:
+        logger.error(
+            "Error parsing XML output: %s - Original Error: %s",
```

Comment:
```suggestion
            "Error parsing XML output: %s - Original Error: %s",
```

Start with the error, then the output, otherwise it will be hard to read.